mongo/client_examples_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+func ExampleConnect_replicaSet() {
+	// Create and connect a Client to a replica set deployment.
+	// Given this URI, the Go driver will first communicate with localhost:27017 and use the response to discover
+	// any other nodes in the replica set.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017/?replicaSet=replset")
+	client, err := mongo.Connect(context.TODO(), clientOpts)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer func() {
+		if err = client.Disconnect(context.TODO()); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	// Call Ping to verify that the deployment is up and the Client was configured successfully.
+	if err = client.Ping(context.TODO(), readpref.Primary()); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func ExampleConnect_sharded() {
+	// Create and connect a Client to a sharded deployment.
+	// The URI for a sharded cluster should specify all mongos nodes in the cluster.
+
+	// Versions 1.1.0 or later of the driver support SRV polling for mongos discovery. Prior to this version, using
+	// an SRV connection string for a sharded cluster was supported, but the hosts were only resolved during initial
+	// connection string parsing. Because individual mongos servers are not aware of each other, added/removed mongos
+	// nodes wouldn't be discovered without an application restart. SRV polling allows the driver to periodically
+	// perform DNS resolution in the background and update the its view of the cluster when nodes are added/removed.
+
+	clientOpts := options.Client().ApplyURI("mongodb://localhost:27017,localhost:27018")
```

Comment:
I'm unclear why you've described SRV polling, but aren't using a "mongodb+srv" connection string?  I think the latter is required for sharded cluster SRV polling, right?

Maybe use this just for non SRV sharded cluster and move your comment to the srv example?

Reply:
I thought the comment about SRV polling belonged in this example because it only applies to sharded clusters, but I can see the reasoning for moving it to the SRV example. Moved in new commit.